xml/System.Runtime.Loader/AssemblyLoadContext.xml

```diff
@@ -20,8 +20,45 @@
+     * The <xref:System.Runtime.Loader.AssemblyLoadContext.Default?displayProperty=nameWithType> represents the runtimes default binding context which is used for the framework and the application main assembly and its normal dependencies.
+     * The <xref:System.Reflection.Assembly.LoadFile(System.String)?displayProperty=nameWithType> method isolates the assemblies it loads by instantiating the most basic <xref:System.Runtime.Loader.AssemblyLoadContext>.  It has a simplistic isolation scheme which loads each assembly in its own <xref:System.Runtime.Loader.AssemblyLoadContext> with no dependency resolution.
+ 
+ To implement the <xref:System.Reflection.Assembly.LoadFrom(System.String)?displayProperty=nameWithType> event, the runtime adds a <xref:System.Runtime.Loader.AssemblyLoadContext.Resolving> event handler to bind to dependent files in the same directory.
+
+### Application usage
+ An application can create its own <xref:System.Runtime.Loader.AssemblyLoadContext> to create a custom solution for advanced binding scenarios. The customization focuses on defining dependency resolution mechanisms.
+ 
+ The <xref:System.Runtime.Loader.AssemblyLoadContext> provides two extension points to implement managed assembly resolution.
+     1. The <xref:System.Runtime.Loader.AssemblyLoadContext.Load(System.Reflection.AssemblyName)?displayProperty=nameWithType> provides the first chance for the <xref:System.Runtime.Loader.AssemblyLoadContext> to resolve the assembly, load it and return it. If the <xref:System.Runtime.Loader.AssemblyLoadContext.Load(System.Reflection.AssemblyName)?displayProperty=nameWithType> returns `null`, the loader will try to load the assembly into the <xref:System.Runtime.Loader.AssemblyLoadContext.Default?displayProperty=nameWithType>.
+     2. If the <xref:System.Runtime.Loader.AssemblyLoadContext.Default?displayProperty=nameWithType> is unable to resolve the assembly, the original <xref:System.Runtime.Loader.AssemblyLoadContext> gets a second chance to resolve the assembly. The runtime raises the <xref:System.Runtime.Loader.AssemblyLoadContext.Resolving> event.
+ 
+ Additionally the <xref:System.Runtime.Loader.AssemblyLoadContext.LoadUnmanagedDll(System.String)?displayProperty=nameWithType> virtual method allows customization of the default unmanaged assembly resolution. The default implementation is often sufficient for most typical scenarios.
+
+### Testing challenges
+ The timing of assembly loading can make testing and debugging difficult. Assemblies are typically loaded without their dependencies immediately being resolved. The dependencies are loaded as they are needed:
+     * When code branches into a dependent assembly
+     * When code loads resources
+     * When code explicitly loads assemblies
+
+### Limitations
+ The runtime's assemblies may only be loaded in the default assembly.  It is not possible to load multiple versions of the runtime in a single process.
```

Comment:
What is a "runtime's assembly"? In reality we only force the 'System.Private.CoreLib' to be a singleton. All other assemblies can in theory be loaded multiple times. I think we should state this clearly here. And add a warning that loading multiple copies or different versions of the framework assemblies can lead to unexpected and hard to diagnose behavior.

Reply:
I am taking a shot at this